src/btree/bt_vrfy.c

```diff
@@ -339,6 +341,104 @@ __verify_addr_ts(WT_SESSION_IMPL *session, WT_REF *ref, WT_CELL_UNPACK *unpack,
     return (0);
 }
 
+/*
+ * __verify_history_key_with_table --
+ *     With a given history store key, and a table name, iterate through the table and check if the
+ *     key exists in the table.
+ */
+static int
+__verify_history_key_with_table(WT_SESSION_IMPL *session, const char *uri, WT_ITEM *hs_key)
+{
+    WT_CURSOR *data_cursor;
+    WT_DECL_RET;
+    WT_ITEM data_key;
+
+    WT_ERR(__wt_open_cursor(session, uri, NULL, NULL, &data_cursor));
+
+    while ((ret = data_cursor->next(data_cursor)) == 0) {
+        /* The cursor raw flag, makes sure that we can place it into a WT_ITEM */
+        F_SET(data_cursor, WT_CURSOR_RAW_OK);
+        WT_ERR(data_cursor->get_key(data_cursor, &data_key));
+        if (__wt_lex_compare(hs_key, &data_key) == 0) {
+            goto done;
```

Comment:
Is there any reason you can't just do `break` here? I don't see a reason for this `done` tag.